### PR TITLE
Fix caret position after inserting completion

### PR DESCRIPTION
**Pull request by CarlRosell**
Fixes: #3834.

### CODE REVIEW

File: app/src/ui/autocompletion/autocompleting-text-input.tsx [hunk 1]
```diff
@@ -295,17 +286,28 @@ export abstract class AutocompletingTextInput<
     const autoCompleteText = autocompletionState.provider.getCompletionText(
       item
     )
+
+    const textWithAutoCompleteText =
+      originalText.substr(0, range.start - 1) + autoCompleteText + ' '
```
Comment (CarlRosell) [inline app/src/ui/autocompletion/autocompleting-text-input.tsx]: This seems to be the current behavior on github.com and the old GitHub client. 
Just adding an extra space right after the inserted completion and position the caret there. Some other applications also do it this way. 
Slack seems to do something a bit less naive by actually looking at the next character and only inserting the extra space if the next character isn't one. 
Facebook will not add anything extra at all.

I chose to do it as the other GitHub clients to provide a unified experience.

Comment (shiftkey) [reply app/src/ui/autocompletion/autocompleting-text-input.tsx]: Yep, makes sense.

File: app/src/ui/autocompletion/autocompleting-text-input.tsx [hunk 2]
```diff
@@ -295,17 +286,28 @@ export abstract class AutocompletingTextInput<
...
+      textWithAutoCompleteText + originalText.substr(range.start + range.length)
+
     element.value = newText
 
     if (this.props.onValueChanged) {
       this.props.onValueChanged(newText)
     }
 
+    // This is pretty gross. Clicking on the list moves focus off the text area.
+    // Immediately moving focus back doesn't work. Gotta wait a runloop I guess?
```
Comment (shiftkey) [inline app/src/ui/autocompletion/autocompleting-text-input.tsx]: This is already an issue with the current app, and I'm not sure of alternatives to fix this.

@niik any other ideas to handle clicking the autocomplete list item stealing focus from the related text element?

Comment (niik) [reply app/src/ui/autocompletion/autocompleting-text-input.tsx]: Not of the top of my head, no. But we don't have to solve that in this PR. But it seems like this PR changes to doing the focus hack regardless of whether the autocomplete happened as the result of a keyboard or pointer action. We should _only_ do this onClick.

Comment (shiftkey) [reply app/src/ui/autocompletion/autocompleting-text-input.tsx]: @niik good shout. I'll see if I can work with @CarlRosell to handle keyboard autocomplete differently to pointer autocompletes :thumbsup:

**Comment by shiftkey** [comment]
@CarlRosell I've added some code to differentiate between keyboard-based autocomplete and mouseclick-based autocomplete, and tidied it up. Feel free to test this and ensure you're happy with the changes, and we can get @niik to sign off on it.

**Comment by CarlRosell** [comment]
Hi @shiftkey and @niik and sorry for my late response. The whole family got the flu and life been chaotic the last two weeks 😅. The changes look good to me and thanks for keeping this alive 👍

**Pull request merged**